src/lib.rs

```diff
@@ -135,13 +134,13 @@ impl EncryptedRecord {
     }
 }
 
-pub struct EnvelopeCipher<S, K, R: SafeRng = ChaChaRng> {
+pub struct EnvelopeCipher<K, S = Aes128Gcm, R: SafeRng = ChaChaRng> {
```

Comment:
Looking at this a bit more closely, I wonder if making the EnvelopeCipher generic on the Cipher type (`S`) is even necessary given the KeyProvider already has it. We could possibly use an associated type pattern?

https://doc.rust-lang.org/beta/rust-by-example/generics/assoc_items/types.html

Reply:
Looks like you're right. I've made the prototype here: https://github.com/cipherstash/envelopers/pull/36
Looks like this is the way to go, so I've merged that branch to be included in this PR directly.